src/pages/index.js

```diff
@@ -17,131 +21,185 @@ import ModalWithForm from "../components/ModalWithForm.js";
 import ModalWithImage from "../components/ModalWithImage.js";
 import Section from "../components/Section.js";
 import Modal from "../components/Modal.js";
-
-// DOM elements
-const cardsWrap = document.querySelector("#cardList");
-const profileEditButton = document.querySelector("#profileEditButton");
-const profileEditModal = document.querySelector("#profileEditModal");
-const profileCloseButton = document.querySelector("#profileCloseButton");
-const profileName = document.querySelector("#profileName");
-const profileDescription = document.querySelector("#profileDescription");
-const profileNameInput = document.querySelector("#profileNameInput");
-const profileDescriptionInput = document.querySelector(
-  "#profileDescriptionInput"
-);
-
-const profileForm = document.querySelector("#profileEditForm");
-const profileAddButton = document.querySelector("#profileAddButton");
-const cardAddButton = document.querySelector("#cardAddButton");
-const cardTemplate = document.querySelector("#cardTemplate");
-const cardAddModal = document.querySelector("#cardAddModal");
-const cardCloseButton = document.querySelector("#cardCloseButton");
-const cardTitleInput = document.querySelector("#cardTitleInput");
-const cardImageInput = document.querySelector("#cardImageInput");
-const cardPreviewModal = document.querySelector("#cardPreviewModal");
-const cardPreviewCloseButton = document.querySelector(
-  "#modalCardPreviewCloseButton"
-);
-
-const cardImage = cardPreviewModal.querySelector("#modalPreviewImage");
-const cardCaption = cardPreviewModal.querySelector("#modalCaption");
-const cardModalButton = cardPreviewModal.querySelector("#cardModalButton");
+import Api from "../utils/Api.js";
+import ModalWithConfirmation from "../components/ModalWithConfirmation.js";
+
+// Create an instance of the API class
+const api = new Api(apiConfig);
+let userId;
+let cardSection;
+
+const cardPreviewModal = new ModalWithImage(selectors.cardPreviewModal);
+cardPreviewModal.setEventListeners();
+
+const deleteForm = new ModalWithConfirmation(selectors.deleteModal, (id) => {
+  deleteForm.setApiCalling(true);
+  api
+    .deleteCard(id)
+    .then(() => {
+      deleteForm.close();
+      cardSection.removeItem(id)
+    })
+    .catch((err) =>
+      console.log(`An error occurred when deleting card: ${err}`)
+    )
+    .finally(() => deleteForm.setApiCalling(false));
+});
+deleteForm.setEventListeners();
+
+// Create a function to create card instances
+const card = (data) => {
```

Comment:
Function names should start with a **verb** in infinitive (there is such a point in the check list). It’s common in coding to do it so that functions could `speak` for themselves about what they do.


`createCard`